python/data_server/dataset_cache.py

```diff
@@ -0,0 +1,42 @@
+import threading
+
+import cachetools
+
+from data_server import gcs_utils
+
+
+class DatasetCache():
+    def __init__(self, max_cache_size=8, cache_ttl=2 * 3600):
```

Comment:
- 2 hrs is a gut-feeling number based on how often I thought it would make sense to refresh and how long I thought an individual instance might run. Right now that is much less than 2 hours, but this is tunable.
- I chose TTL because the data does get updated periodically and we don't want to be serving stale data indefinitely. Depending on how our performance is, we might choose to decrease the TTL to keep data fresher.
- I thought about this also, and think that it will make sense to count bytes rather than num elements in the future. I think that we might want to consider frequency-based solutions as well.

Basically this is a middle ground first attempt that we can iterate on based on how our needs shake out.

Reply:
Oh whoops I forgot about the data being updated haha, it definitely needs a TTL

Yeah that all sounds good!